src/org/stringtemplate/v4/misc/AggregateModelAdaptor.java

```diff
@@ -29,17 +29,17 @@
 package org.stringtemplate.v4.misc;
 
 import org.stringtemplate.v4.Interpreter;
+import org.stringtemplate.v4.ModelAdaptor;
 import org.stringtemplate.v4.ST;
 
-import java.util.Map;
-
 /** Deal with structs created via {@link ST#addAggr}{@code ("structname.{prop1, prop2}", ...);}. */
-public class AggregateModelAdaptor extends MapModelAdaptor {
+public class AggregateModelAdaptor implements ModelAdaptor<Aggregate> {
+    private final MapModelAdaptor mapAdaptor = new MapModelAdaptor();
```

Comment:
```suggestion
    private static final MapModelAdaptor mapAdaptor = new MapModelAdaptor();
```

Reply:
I intentionally did not make it static to follow the pattern of all adaptors, in that they are instance-only. I doubt efficiency will be measurably affected, since `AggregateModelAdaptor` is usually instantiated only once per group. Going this route, all Adaptors (and Renderers) should be singletons.